torch/fx/tensor_type.py

```diff
@@ -0,0 +1,62 @@
+class TensorType:
+    """
+    Tensor_Type defines a type for tensors, which consists of a list of dimensions.
+    Example:
+        class M(torch.nn.Module):
+            def forward(self, x:Tensor_Type((1,2,3, Dyn)), y:Tensor_Type((1,2,3, Dyn))):
+                return torch.add(x, y)
+    """
+
+    def __init__(self, dim):
+        self.__origin__ = TensorType
+        self.__args__ = dim
+
+    def __repr__(self):
+        return f'Tensor_Type[{self.__args__}]'
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            return list(self.__args__) == list(other.__args__)
+        else:
+            return False
+
+    @staticmethod
+    def __class_getitem__(*args):
+        assert isinstance(args[0], tuple)
+        return TensorType(args[0])
+
+
+class _DynType:
+    """
+    _DynType defines a type which stands for the absence of type information.
+    """
+    def __init__(self):
+        self.__name__ = '_DynType'
```

Comment:
Is this for printing it out or something else?

I think the more canonical way of defining the printouts is to define a `__repr__` method

Reply:
if I remove the name attribute the symbolic execution complains 
`AttributeError: '_DynType' object has no attribute '__name__'`

Reply:
Can you post the full error? (including stack trace)

That probably indicates that the prettyprint process for types is expecting that attribute

Reply:
Error
Traceback (most recent call last):
  File "/Users/migeedz/pytorch/test/fx/test_gradual_type.py", line 38, in test_annotate
    symbolic_traced : torch.fx.GraphModule = symbolic_trace(module)
  File "/Users/migeedz/pytorch/torch/fx/_symbolic_trace.py", line 863, in symbolic_trace
    return GraphModule(tracer.root, graph, name)
  File "/Users/migeedz/pytorch/torch/fx/graph_module.py", line 234, in __init__
    self.graph = graph
  File "/Users/migeedz/pytorch/torch/nn/modules/module.py", line 1179, in __setattr__
    object.__setattr__(self, name, value)
  File "/Users/migeedz/pytorch/torch/fx/graph_module.py", line 259, in graph
    self.recompile()
  File "/Users/migeedz/pytorch/torch/fx/graph_module.py", line 467, in recompile
    python_code = self._graph.python_code(root_module='self')
  File "/Users/migeedz/pytorch/torch/fx/graph.py", line 803, in python_code
    return self._python_code(root_module, namespace)
  File "/Users/migeedz/pytorch/torch/fx/graph.py", line 955, in _python_code
    emit_node(node)
  File "/Users/migeedz/pytorch/torch/fx/graph.py", line 897, in emit_node
    maybe_type_annotation = '' if node.type is None else f' : {type_repr(node.type)}'
  File "/Users/migeedz/pytorch/torch/fx/graph.py", line 854, in type_repr
    args = [type_repr(arg) for arg in o.__args__]
  File "/Users/migeedz/pytorch/torch/fx/graph.py", line 854, in <listcomp>
    args = [type_repr(arg) for arg in o.__args__]
  File "/Users/migeedz/pytorch/torch/fx/graph.py", line 859, in type_repr
    return add_global(typename, o)
  File "/Users/migeedz/pytorch/torch/fx/graph.py", line 826, in add_global
    return _get_qualified_name(obj)
  File "/Users/migeedz/pytorch/torch/fx/node.py", line 64, in _get_qualified_name
    if getattr(builtins, func.__name__, None) is func:
AttributeError: '_DynType' object has no attribute '__name__'